src/prometheus-stack/chart/templates/uds-rules.yaml

```
@@ -0,0 +1,55 @@
+# Copyright 2025 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+{{- if .Values.clusterHealthRules.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: "{{ .Release.Name }}-uds-cluster-health"
+  namespace: {{ .Release.Namespace }}
+  labels:
+    release:  {{ .Release.Name }}
+spec:
+  groups:
+    - name: "{{ .Release.Name }}-uds-cluster-health.rules"
+      interval: 1m
+      rules:
+        - record: uds_cluster_health_score
+          expr: |
+            (
+              # 25% weight for API Server health
+              (avg(up{job="apiserver"}) or vector(1)) * 25
+              +
+              # 25% weight for Kubelet health
+              (avg(up{job="kubelet"}) or vector(1)) * 25
+              +
+              # 15% weight for CoreDNS health
+              (avg(up{job="coredns"}) or vector(1)) * 15
+              +
+              # 15% weight for Pepr health
+              (avg(up{job="pepr-uds-core"}) or vector(1)) * 15
+              +
+              # 10% weight for Keycloak health
+              (avg(up{job="keycloak-http"}) or vector(1)) * 10
+              +
+              # 10% weight for Prometheus health
+              (avg(up{job="kube-prometheus-stack-prometheus"}) or vector(1)) * 10
+            )
+        - alert: UDSClusterHealthScoreWarning
+          expr: uds_cluster_health_score < 95
```

I think this will also fire if it is below 90% creating duplicate alerts. consider making the expression: 
```suggestion
          expr: uds_cluster_health_score < 95 and uds_cluster_health_score >= 90
```